protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -34,22 +74,51 @@ use crate::Error::MissingBytes;
 #[cfg(feature = "noise_sv2")]
 use crate::State;
 
-#[cfg(feature = "noise_sv2")]
-pub type StandardNoiseDecoder<T> = WithNoise<Buffer, T>;
+/// An encoded or decoded frame that could either be a regular or Noise-protected frame.
 pub type StandardEitherFrame<T> = Frame<T, <Buffer as IsBuffer>::Slice>;
+
+/// An encoded or decoded Sv2 frame.
 pub type StandardSv2Frame<T> = Sv2Frame<T, <Buffer as IsBuffer>::Slice>;
+
+/// Standard decoder with Noise protocol support.
+#[cfg(feature = "noise_sv2")]
+pub type StandardNoiseDecoder<T> = WithNoise<Buffer, T>;
+
+/// Standard Sv2 decoder without Noise protocol support.
 pub type StandardDecoder<T> = WithoutNoise<Buffer, T>;
 
+/// Decoder for Sv2 frames with Noise protocol support.
 #[cfg(feature = "noise_sv2")]
 pub struct WithNoise<B: IsBuffer, T: Serialize + binary_sv2::GetSize> {
+    /// Used to maintain type information for the generic parameter `T`, which represents the type
+    /// of frames being decoded.
+    ///
+    /// `T` refers to a type that implements the necessary traits for serialization
+    /// (`binary_sv2::Serialize`), deserialization (`binary_sv2::Deserialize`), and size
+    /// calculation (`binary_sv2::GetSize`).
```

Comment:
private fields are never exposed to reader eyes

it's good to document this for maintainer eyes, but we should use Non Doc comments, as proposed in https://github.com/stratum-mining/stratum/discussions/970#discussioncomment-10697537